### ISSUE TITLE
Suggestion: add option to link Audio and Microphone devices

### ISSUE THREAD

**Issue by AlkoSoul** [xenolightning/AudioSwitcher_v1#1969]
It's pretty simple really (as the title suggests), why not add the ability to use a keyboard shortcut to set BOTH the audio output AND the microphone input.

For example, when no-one's home, I use my PC's 5.1 surround system speakers, and the Mic on my webcam.  When I switch to my USB headphones, I use them for both audio and microphone, as they have both.

Currently I have to use a total of 4 separate keyboard shortcuts the switch between the two, when realistically, I only need two. ;)

P.s. Love the software so far and donated a little while ago - the default Windows 8 / 8.1 / 10 setup is a bit on the shite side - your software makes it easy :D

**Comment by xenolightning** [comment]
Thanks! Yeah I'm not sure why Windows makes such a simple task so obscure.

Yep totally valid request, and is "impossible" to do int v1.x due to how the hot keys work.

Each hotkey is hard bound to a single hardware device, which I admit is a terrible implementation, it just doesn't warrant changing it when 2.0 is somewhere on the horizon.

What this really comes under is a "profile", where you have a hotkey to switch profiles.

Can be achieved in 2.0, using a script, which can set defaults, and/or volume levels (really whatever you want). You can then run a script on a hotkey press.

Here's an example script, in the audio switcher script editor, which does that. It set the defaults then shows a little balloon notification in the tray area.

![script](https://cloud.githubusercontent.com/assets/6000196/13270190/4935475e-daf1-11e5-812f-6e769019c820.png)


There was a feature request somewhere for exactly this, not sure where it went though so I'll mark this one, and keep it open.

**Comment by AlkoSoul** [comment]
Hi, and thanks for the response :)

I wasn't trying to suggest that you "fix" the V1 switcher, just a suggestion for the next update (didn't realise V2 was on it's way!)

I'm not much for programming or coding - hence why I tried Audio Switcher, and donated within 10 minutes of setting it up - it has saved me a lot of hassle so far.

As far as my request goes, I would simply say that you need to elevate the keyboard shortcut by one level.  I.e it runs a simple script that activates the speakers and microphone in the same way that the individual shortcuts work in V1 (effectively running two "keyboard shortcuts" via only one).  I realise that this sounds easy, but probably isn't in code, but I look forward to V2 and giving it a damn good testing ;)

**Comment by xenolightning** [comment]
Yep, you're totally on point.

The hotkeys in 2.0 now do "actions" instead of just setting default devices.

I've created three actions so far:
- Cycle through favourite devices
- Set Defaults
- Run a script

With the possibly for more to come in the future. It's a much better system, and is significantly more robust.

The scripts are aimed at "power users", where they want really specific functionality on case by case basis, which doesn't fit into the core software.

**Comment by groove200** [comment]
Cant wait for 2.0

Donate button click waiting ;-)

**Comment by malikoth** [comment]
I'm not sure if this comment belongs here, or in a new issue.  Let me know what you think.

I'd kill for the ability to click on the tray icon and have it cycle through my favorite speaker devices AND my favorite recording devices.  I don't know if this is more doable than the keyboard shortcuts, since this is talking about a mouse event.

In the event that the best answer is "wait for v2", then I would just like to suggest that you ensure v2 has the ability to let you switch profiles by clicking on the tray icon.

**Comment by xenolightning** [comment]
@malikoth Here's a good as place as any.

It's certainly a v2 feature.

In v2, there's a handful of "events" that you can "bind" to and do anything you want.

Quick Summary of events that exist:
- Exe opening/closing
- System Events - Default Device Changing etc..
- Hotkeys
- Tray Icon Events. Left/Middle click, including Crtl/Shift/Alt modifiers

**Comment by xenolightning** [comment]
Related: https://github.com/xenolightning/AudioSwitcher_v1/issues/2131

**Comment by singlemalt8** [comment]
Joining this conversation late, but huge thanks to Sean for all his hard work on this - an invaluable tool that Windows somehow forgot about!

I know you are working on v2.0 (can't wait to see that!) but wanted to echo that the ability to "link" playback and recording devices in the quick-switch and hotkeys mode is high value for heavy telephony users.  Moving between speakerphones and headsets requires pairing the playback and recording devices, so would love the ability to link them for seamless switching.

Again - awesome work and thanks for all your "spare time" efforts!

**Comment by speed-of-heat** [comment]
yes please; i only found this app a few days ago and, it is a wonderful improvement, but, I really look forward to the version that enables input and output to be switched together!

**Comment by Gadagadanagadanagana** [comment]
Is there maybe any means for triggering both hotkeys with one key? Like a really tiny script? It does not look as if this project will be continued.
